tests/unit/StoreCurrentAccount.spec.ts

```diff
@@ -1,47 +1,32 @@
-import { getVuexStore } from '@/store'
+import { createPinia, setActivePinia } from 'pinia'
+import { useAuthenticationStore } from '@/stores'
 import { shallowMount, createLocalVue } from '@vue/test-utils'
-import Vue, { nextTick } from 'vue'
+import Vue from 'vue'
 import Vuetify from 'vuetify'
 
 Vue.use(Vuetify)
 const localVue = createLocalVue()
+setActivePinia(createPinia())
+const authenticationStore = useAuthenticationStore()
 
 describe('testing current account module', () => {
-  const dummyAccount = (accountType: string) => {
-    return {
-      accountStatus: 'string',
-      accountType: accountType,
-      additionalLabel: 'string',
-      id: 1234,
-      label: 'string',
-      productSettings: 'string',
-      type: 'string',
-      urlorigin: 'string',
-      urlpath: 'string'
-    }
-  }
-
   it('fetches current account from session storage - PREMIUM', async () => {
     jest.isolateModules(() => async () => {
       const SbcHeader = require('sbc-common-components/src/components/SbcHeader.vue').default
-      const store = await getVuexStore() as any // remove typings for unit tests
-      shallowMount(SbcHeader, { store, localVue })
+      shallowMount(SbcHeader, { localVue })
 
-      store.commit('account/setCurrentAccount', dummyAccount('PREMIUM'), { root: true })
-      expect(store.getters.isPremiumAccount).toBe(true)
-      expect(store.getters.isSbcStaff).toBe(false)
+      expect(authenticationStore.isPremiumAccount).toBe(true)
+      expect(authenticationStore.isSbcStaff).toBe(false)
     })
   })
 
   it('fetches current account from session storage - SBC_STAFF', async () => {
     jest.isolateModules(() => async () => {
       const SbcHeader = require('sbc-common-components/src/components/SbcHeader.vue').default
-      const store = await getVuexStore() as any // remove typings for unit tests
-      shallowMount(SbcHeader, { store, localVue })
+      shallowMount(SbcHeader, { localVue })
 
-      store.commit('account/setCurrentAccount', dummyAccount('SBC_STAFF'), { root: true })
-      expect(store.getters.isPremiumAccount).toBe(false)
-      expect(store.getters.isSbcStaff).toBe(true)
+      expect(authenticationStore.isPremiumAccount).toBe(false)
+      expect(authenticationStore.isSbcStaff).toBe(true)
```

Comment:
OK, wait. These tests should still not be passing since they have the same setup but test opposite things.

I think this is a blocker (it points to something wrong, somewhere).

Reply:
So, regarding the Authentication Store, between all the getters, I only see one being used and that's the `getKeycloakGuid` getter. Why are the other getters there?

Also, there's an another getter that checks if user is staff that's being used and it's in the rootStore (`isRoleStaff`). Should I instead use it in this test file then?

Reply:
Jonathan probably added the other getters for completeness / future use.

It would be ideal if the unit tests ran the same code as the user (runtime). I don't know what's needed to initialize the auth store correctly in the tests.

Reply:
Ohh OK that makes sense. I was wondering why were these getters there. I'll look into this Sev and try to change it to test the correct stuff.

Reply:
I decided to skip this test and returned Jonathan's code since I found out that the `SbcHeader` shared component still uses Vuex. Let me know what you think Sev.

Reply:
I think that we can leave this until later as long as the code works correctly locally.

Reply:
I totally agree. The code runs locally well. I was thinking of removing the authentication store and moving the only getter we use from it to the root store. Or, we can handle this later in a later ticket. Thoughts?

Reply:
My thoughts:
- change no more than needed in this ticket
- ok to disable unit tests temporarily if the code runs fine (local or dev)
- keep track of things we need to go back and fix/update

So, I have a slight preference for keeping Jonathan's authentication store code.

Reply:
Alright, noted! Makes total sense. I'll add a checkbox for this in the fix unit tests ticket.

Reply:
OK, so these tests are partly updated for Pinia but disabled as they still don't pass.

Given that the code works locally and only the unit test fails (when enabled), and we've planned the follow-on ticket for the new sprint, then I'm OK with this. Resolving.